core/src/databases/database.rs

```diff
@@ -83,16 +85,216 @@ impl Database {
                 .into_iter()
                 .collect::<Vec<_>>();
 
-                Ok(DatabaseSchema(
-                    rows.into_par_iter()
-                        .map(|(table, rows)| {
-                            Ok((
-                                table.table_id().to_string(),
-                                DatabaseSchemaTable::new(table, TableSchema::from_rows(&rows)?),
-                            ))
-                        })
-                        .collect::<Result<HashMap<_, _>>>()?,
-                ))
+                let schema = rows
+                    .par_iter()
+                    .map(|(table, r)| {
+                        Ok((
+                            table.table_id().to_string(),
+                            DatabaseSchemaTable::new(table.clone(), TableSchema::from_rows(&r)?),
+                        ))
+                    })
+                    .collect::<Result<HashMap<_, _>>>()?;
+
+                let returned_rows = match return_rows {
```

Comment:
Can we move this in a different function.

I undersstand that you are looking to save a load of the rows but really eventually the schema will be stored and won't require loading the rows. Fine to load the rows twice in the experimental non optimized phase. We'll already be loading them way too many time due to the non cached version of the schema


Interfaces matter more than implementation especially if the reason you're poking a hole is a transient one.

Reply:
Ok

Reply:
Done